weather_forecast_retrieval/hrrr_preprocessor.py

```diff
@@ -141,6 +176,9 @@ def run(self):
 
             self.call_wgrib2(crop_action)
 
+            # Check that the file has been created and is not empty or corrupt
+            self.check_for_good_file(new_hrrr_file)
+
```

Comment:
Wouldn't this check be better done first before attempting a subset and extraction of variables?
If they are missing I would suspect the input files to be corrupt. Then you can spare yourself the previous two commands.

Reply:
My thought was to check the final file that we'd be using. But you're probably right that the if the variables aren't there then cropping won't really matter. If we combine like you mention below, it'll have to be at the end!